erpnext/schools/doctype/student_group/student_group.py

```diff
@@ -10,47 +10,64 @@
 from erpnext.schools.api import get_student_batch_students
 
 class StudentGroup(Document):
-	def autoname(self):
-		self.name = frappe.db.get_value("Course", self.course, "course_abbreviation")
-		if not self.name:
-			self.name = self.course
-		if self.student_batch:
-			self.name += "-" + self.student_batch
-		else:
-			prog_abb = frappe.db.get_value("Program", self.program, "program_abbreviation")
-			if not prog_abb:
-				prog_abb = self.program
-			if prog_abb:
-				self.name += "-" + prog_abb
-			if self.academic_year:
-				self.name += "-" + self.academic_year
-		if self.academic_term:
-			self.name += "-" + self.academic_term
-	
 	def validate(self):
+		self.validate_mandatory_fields()
 		self.validate_strength()
-		self.validate_student_name()
-		self.validate_name()
-		if self.student_batch:
-			self.validate_student_batch()
+		self.validate_students()
 		validate_duplicate_student(self.students)
 
+	def validate_mandatory_fields(self):
+		if self.group_type == "Course" and not self.course:
```

Comment:
See, thats why I told to rename the field to "Group Based On". Here, you have used "Course", instead of "Course wise".